starter-code/routes/logIn.js

```diff
@@ -0,0 +1,45 @@
+const express = require("express");
+const router = express.Router();
+
+const User = require("../user/user.js");
+
+const bcrypt         = require("bcrypt");
+const bcryptSalt     = 10;
+
```

Comment:
these routes should have a `router.use(...)` block at the top of the file, defining a middleware that runs before the routes

```
router.use(function (req, res, next) {
  if (!req.session.currentUser) {
    next(); // we DO NOT have a user in the session, continue executing routes
  } else {
    res.redirect('/welcome'); // user is logged in, redirect to /welcome
  }
});
```